### PR TITLE
Feature: Add the ability to configure the SA account lifetime

### PR DESCRIPTION
**Pull request by mabhi**
Implements the feature : - [98](https://github.com/paralus/paralus/issues/98)

Signed-off-by: mabhi <abhijit.mukherjee@infracloud.io>

### What does this PR change?

- Add the ability to configure the SA account lifetime. This also handles persisting new organisation setting for SA de-provision duration, used by relay agent during cleanup of stale resources

### Does the PR depend on any other PRs or Issues? If yes, please list them.

- No

### Checklist

I confirm, that I have...

- [x] Read and followed the contributing guide in `CONTRIBUTING.md`
- [ ] Added tests for this PR
- [x] Formatted the code using `go fmt` (if applicable)
- [ ] Updated [documentation on the Paralus docs site](https://github.com/paralus/website/blob/main/docs) (if applicable)
- [ ] Updated `CHANGELOG.md`

### CODE REVIEW

File: api/def/clients/sentry/client/audit_information/audit_information_client.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ type ClientService interface {
 }
 
 /*
-  AuditInformationLookupCluster audit information lookup cluster API
```
Comment (niravparikh05) [inline api/def/clients/sentry/client/audit_information/audit_information_client.go:39]: why do we have these formatting changes ?

File: server/kubeconfig.go [hunk 2]
```diff
@@ -101,18 +101,24 @@ func (s *kubeConfigServer) GetOrganizationSetting(ctx context.Context, req *sent
...
+	sessData, ok := service.GetSessionDataFromContext(ctx)
+	if !ok {
+		return nil, fmt.Errorf("unable to retrieve session data")
+	}
+
 	if orgID != opts.Organization {
 		opts.Organization = orgID
 	}
-	ks, err := s.kss.Get(ctx, opts.Organization, "", false)
+	ks, err := s.kss.Get(ctx, opts.Organization, sessData.Account, false)
```
Comment (niravparikh05) [inline server/kubeconfig.go:112]: why are we passing account here? this should be organization specific setting and not account specific ..

File: server/kubeconfig.go [hunk 3]
```diff
@@ -101,18 +101,24 @@ func (s *kubeConfigServer) GetOrganizationSetting(ctx context.Context, req *sent
...
 	if err == constants.ErrNotFound {
-		return &sentryrpc.GetKubeconfigSettingResponse{ValiditySeconds: 31536000}, nil
+		return &sentryrpc.GetKubeconfigSettingResponse{ValiditySeconds: 31536000, SaValiditySeconds: 28800}, nil
```
Comment (niravparikh05) [inline server/kubeconfig.go:114]: there should be specific validation for mix, max sa validaty seconds..

File: server/kubeconfig.go [hunk 4]
```diff
@@ -160,8 +167,9 @@ func (s *kubeConfigServer) UpdateOrganizationSetting(ctx context.Context, req *s
 	err = s.kss.Patch(ctx, &sentry.KubeconfigSetting{
 		OrganizationID:              opts.Organization,
 		PartnerID:                   opts.Partner,
-		AccountID:                   "",
+		AccountID:                   opts.Account,
```
Comment (niravparikh05) [inline server/kubeconfig.go:170]: not required, as we are currently in organization context

File: internal/models/kubeconfig_setting.go [hunk 5]
```diff
@@ -16,6 +16,7 @@ type KubeconfigSetting struct {
 	AccountId                   uuid.UUID `bun:"account_id,type:uuid,notnull"`
 	Scope                       string    `bun:"scope,notnull"`
 	ValiditySeconds             int64     `bun:"validity_seconds,notnull,default:0"`
+	SaValiditySeconds           int64     `bun:"sa_validity_seconds,notnull,default:0"`
```
Comment (niravparikh05) [inline internal/models/kubeconfig_setting.go:19]: we decided the default will be eight hours right? so it should be the corresponding value in seconds.

File: pkg/sentry/authz/authz.go [hunk 6]
```diff
@@ -321,9 +328,16 @@ func GetAuthorization(ctx context.Context, req *sentryrpc.GetUserAuthorizationRe
 	accountID := cnAttr.AccountID
 	orgID := cnAttr.OrganizationID
 	partnerID := cnAttr.PartnerID
+	ks, err := kss.Get(ctx, orgID, accountID, cnAttr.IsSSO)
```
Comment (niravparikh05) [inline pkg/sentry/authz/authz.go:331]: so, this is an org level setting and not user level

**Review by niravparikh05 — CHANGES_REQUESTED** [review]

**Comment by mabhi** [comment]
Cancelling the pull request as opened another one

**Pull request closed without merging**